### ISSUE TITLE
False positive unnecessary-dict-index-lookup on del

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#6645]
### Bug description
`test.py`:
```python3
d = {0: 1}
for (key, value) in d.items():
    del d[key]
```
This code is not fully correct, as it's not recommended to modify dictionary while iterating.

However, the error `pylint` is giving is also incorrect.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:3: [R1733(unnecessary-dict-index-lookup), ] Unnecessary dictionary index lookup, use 'value' instead
```

### Expected behavior

No problem reported.

Or maybe a warning about modifying an iterable while iterating.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
```

### OS / Environment

```shell
$ cat /etc/issue
Ubuntu 20.04.4 LTS \n \l

$ uname -a
Linux ubuntu 5.4.0-110-generic #124-Ubuntu SMP Thu Apr 14 19:46:19 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by jacobtylerwalls** [comment]
Hi @jolaf thanks for the suggestion. For a modifying while iterating check we have #2471. For this specific issue with the `unnecessary-dict-index-lookup`, I guess I don't see a problem with it. The dict isn't being altered.

**Comment by jolaf** [comment]
Unfortunately, I can't reproduce the problem now, while I could in the past. :(

The problem definitely appears in my project code, but that's too complex to post. I'll try to create some middle-ground test less complex than my project code, but still reproducing the problem reliably.

**Comment by jolaf** [comment]
I fully agree on the latter.
